engine/src/main/scala/cromwell/engine/language/LanguageConfiguration.scala

```diff
@@ -0,0 +1,50 @@
+package cromwell.engine.language
+
+import java.util.Map.Entry
+
+import com.typesafe.config.ConfigFactory
+import cromwell.engine.language.CromwellLanguages.{CromwellLanguageName, CromwellLanguageVersion}
+
+import scala.collection.JavaConverters._
+
+case class LanguageConfigurationEntry(name: CromwellLanguageName, versions: Map[CromwellLanguageVersion, String])
+
+
+object LanguageConfiguration {
+  private val LanguagesConfig = ConfigFactory.load.getConfig("languages")
+  private val LanguageNames: Set[String] = LanguagesConfig.entrySet().asScala.map(findFirstKey).toSet
+
+  val AllLanguageEntries: List[LanguageConfigurationEntry] = LanguageNames.toList map { languageName =>
+
+    val languageConfig = LanguagesConfig.getConfig(languageName)
+    val versionSet = languageConfig.getConfig("versions")
+    val languageVersionNames: Set[String] = versionSet.entrySet().asScala.map(findFirstKey).toSet
+
+    val versions = (languageVersionNames.toList map { languageVersionName =>
+      val entry = versionSet.getConfig(s""""$languageVersionName"""")
```

Comment:
am I missing something or is this an elaborately roundtripped String?

Reply:
it's adding `"` to the start and end... elaborately